module/data/item/spell.mjs

```diff
@@ -295,6 +295,31 @@ export default class SpellData extends ItemDataModel.mixin(ActivitiesTemplate, I
     return 1;
   }
 
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  get scalingIncrease() {
+    const actor = this.parent.actor;
+    if ( (this.level !== 0) || !actor ) return null;
+    let level = actor.system.details?.level;
+    if ( !level && (actor.type === "npc") ) {
+      if ( this.preparation.mode === "innate" ) level = Math.ceil(actor.system.details.cr);
```

Comment:
We should add a getter for relevant level property (`details.level` and `details.cr`) on the actor system data models, then this would play nicer with module subtypes.

Reply:
Hmm, that was my original design but NPC casting depends on the preparation mode of the cantrip, at least according to the old logic. I wonder if it would make sense to depart from the previous cantrip scaling behavior and just use something like `this.details.level ?? this.details.spellLevel ?? this.details.cr` (to take NPCs with classes into account also).

Reply:
Innate Spellcasting on monsters must use their CR for scaling according to the rules in the Monster Manual. Using their configured spellcaster level would be incorrect.

Reply:
Fixed, using a helper method now rather than a getter.